deps/rabbit/src/rabbit_table.erl

```diff
@@ -53,8 +52,14 @@ create(TableName, TableDefinition) ->
             throw({error, {table_creation_failed, TableName, TableDefinition1, Reason}})
     end.
 
--spec exists(mnesia_table()) -> boolean().
+-spec exists(mnesia:table()) -> boolean().
 exists(Table) ->
+    case mnesia:is_transaction() of
+        true ->
+            mnesia_schema:get_tid_ts_and_lock(schema, read);
```

Comment:
FTR, this function hasn't really changed in 13 years, so we can assume this "semi-private" API element is here to stay for a few more years before we move entirely to Khepri.